tests/test_core.py

```diff
@@ -175,10 +187,23 @@ def test_databricks_cli_credential_provider_installed_legacy(config, monkeypatch
     assert databricks_cli(config) == None
 
 
-def test_databricks_cli_credential_provider_installed_new(config, monkeypatch, tmp_path):
+def test_databricks_cli_credential_provider_installed_new(config, monkeypatch, tmp_path, mocker):
+    get_mock = mocker.patch('databricks.sdk.credentials_provider.CliTokenSource.refresh',
+                            return_value=Token(access_token='token',
+                                               token_type='Bearer',
+                                               expiry=datetime(2023, 5, 22, 0, 0, 0)))
     write_large_dummy_executable(tmp_path)
-    monkeypatch.setenv('PATH', str(os.pathsep).join([tmp_path.as_posix(), os.environ['PATH']]))
+    path = str(os.pathsep).join([tmp_path.as_posix(), os.environ['PATH']])
+    if platform.system() == 'Windows':
+        path = pathlib.PureWindowsPath(path)
```

Comment:
Are this and line 201 needed? I wonder because `pathlib.Path` should behave in a system-dependent way, i.e. when run on windows they will use windows path semantics and on mac/unix they will use mac/unix path semantics.

Reply:
Thank you for pointing this out. I've addressed the changes accordingly.